src/Orleans/Core/IInternalGrainFactory.cs

```diff
@@ -47,5 +47,21 @@ TGrainInterface GetSystemTarget<TGrainInterface>(GrainId grainId, SiloAddress de
         /// A reference to <paramref name="grain"/> which implements <paramref name="interfaceType"/>.
         /// </returns>
         object Cast(IAddressable grain, Type interfaceType);
+
+        /// <summary>
+        /// Gets a reference to the grain with the provided id.
+        /// </summary>
+        /// <typeparam name="TGrainInterface">The grain reference interface type.</typeparam>
+        /// <param name="grainId">The grain id.</param>
+        /// <returns>A reference to the grain with the provided id.</returns>
+        TGrainInterface GetGrain<TGrainInterface>(GrainId grainId);
```

Comment:
Still, later on I saw usages of this not just for grains, but extensions and system targets... maybe it's better to remove the word Grain from the method name anyway